test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -0,0 +1,1215 @@
+    let selected = [...options];
+
+    // NOTE: starting with selected options already!
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{options}}
+        @isReadOnly={{true}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    assert.dom('[data-multiselect-remove-option]').doesNotExist();
+  });
+
+  test('it uses the results from `@onFilter` to populate the filtered options', async function (assert) {
+    assert.expect(5);
+
+    let selected = ['blue'];
+
+    let handleFilter = (value: unknown) => {
+      assert.strictEqual(
+        value,
+        'y',
+        'Expected the input to match what was entered via fillIn'
+      );
+      assert.step('onFilter');
+
+      return ['yellow'];
+    };
+
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{testColors}}
+        @onFilter={{handleFilter}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await fillIn('[data-multiselect]', 'y');
+
+    assert.verifySteps(['onFilter']);
+
+    assert.dom('[role="option"]').exists({ count: 1 });
+    assert.dom('[role="option"]').hasText('yellow');
+  });
+
+  test('it sets the "active" item to the first one in the list when the input gains focus', async function (assert) {
+    let selected = ['blue'];
+
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it sets the "active" item to the next item in the list when using the DOWN arrow', async function (assert) {
+    let selected = ['blue'];
+
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
```

Comment:
On this test and the one above, it would be helpful if there was a comment / assert about what's showing up as options (I can't tell offhand reading the code) before the key presses are made, i.e `// Current options are "blue", "red"`

Reply:
Ah I see what you mean.  I have a few tests above asserting the contents of the options, but this test is focused solely on the keyboard functionality between items.  I don't necessarily _care_ about the contents or how they are rendered, as another test covers that, but I _do_ care that there are options so we can use our keyboard navigation. 
 Not sure if that's helpful or not, but it was an explicit decision not to re-assert options here but focus on keyboard interaction only.  If you have strong feelings about me re-asserting the contents, I can add it though.  Let me know!

Reply:
That's a good point, makes the test easier to read.